File: benchmark/comparison/results/blind_ts/23.ts
type Tree =
  | { type: "Leaf"; value: number }
  | { type: "Node"; left: Tree; right: Tree };

function Leaf(v: number): Tree { return { type: "Leaf", value: v }; }
function Node(left: Tree, right: Tree): Tree { return { type: "Node", left, right }; }

function depth(t: Tree): number {
  if (t.type === "Leaf") return 1;
  return 1 + Math.max(depth(t.left), depth(t.right));
}

console.log(depth(Leaf(1)));
console.log(depth(Node(Leaf(1), Leaf(2))));
console.log(depth(Node(Node(Leaf(1), Leaf(2)), Leaf(3))));

File: benchmark/comparison/results/blind_py/23.py
class Leaf:
    def __init__(self, v):
        self.v = v

class Node:
    def __init__(self, left, right):
        self.left = left
        self.right = right

def depth(t):
    if isinstance(t, Leaf):
        return 1
    return 1 + max(depth(t.left), depth(t.right))

print(depth(Leaf(1)))
print(depth(Node(Leaf(1), Leaf(2))))
print(depth(Node(Node(Leaf(1), Leaf(2)), Leaf(3))))
